lasagne/layers/dense.py

```diff
@@ -89,6 +88,35 @@ def get_output_for(self, input, **kwargs):
         return self.nonlinearity(activation)
 
 
+class NonlinearityOnlyLayer(Layer):
+    """
+    A layer that just performs a nonlinearity.
+    This is for use in conjunction with MultipleInputsLayer subclasses,
+    which just add or concatenate outputs from layers.  For example, one
+    might want to take inputs of length n and m, perform linear transformations
+    on them using DenseLayer objects, then add the transformed outputs with an
+    ElemwiseSumLayer, and then use this class to apply a nonlinearity
+    (note that this would be different from applying the nonlinearity and
+    then adding).  Alternatively, one could use a ConcatLayer followed by
+    a DenseLayer, but concatenation is less efficient than adding.
```

Comment:
While the explanation is perfect to motivate this PR, I think it's a bit out of scope for the class docstring. I would just write `A layer that just applies a nonlinearity.`, and not give any use case, because there are plenty. The explanation of how to replace a ConcatLayer -> DenseLayer with DenseLayer -> ElementwiseSumLayer would be good for the (not yet existing) tutorial, though.
And last but not least, out of curiosity, have you actually profiled which one is faster? The first one involves a copy operation, but then allows a single dot product, which could actually be faster on GPU than adding up multiple dot products.